test/tests.js

```diff
@@ -0,0 +1,298 @@
+        window.Leanplum.eventProperties[0].value.should.equal(200);
+
+        done();
+    });
+
+    it('should log page view', function(done) {
+        mParticle.forwarder.process({
+            EventDataType: MessageType.PageView,
+            EventName: 'test name',
+            EventAttributes: {
+                attr1: 'test1',
+                attr2: 'test2'
+            }
+        });
+
+        window.Leanplum.trackCustomEventCalled.should.equal(true);
+        window.Leanplum.trackCustomName.should.equal('Viewed test name');
+        window.Leanplum.eventProperties[0].attr1.should.equal('test1');
+        window.Leanplum.eventProperties[0].attr2.should.equal('test2');
+
+        done();
+    });
+
+    it('should log commerce event', function(done) {
+        mParticle.forwarder.process({
+            EventDataType: MessageType.Commerce,
+            ProductAction: {
+                ProductActionType: ProductActionType.Purchase,
+                ProductList: [
+                    {
+                        Sku: '12345',
+                        Name: 'iPhone 6',
+                        Category: 'Phones',
+                        Brand: 'iPhone',
+                        Variant: '6',
+                        Price: 400,
+                        TotalAmount: 400,
+                        CouponCode: 'coupon-code',
+                        Quantity: 1
+                    }
+                ],
+                TransactionId: 123,
+                Affiliation: 'my-affiliation',
+                TotalAmount: 450,
+                TaxAmount: 40,
+                ShippingAmount: 10,
+                CouponCode: null
+            }
+        });
+
+        window.Leanplum.trackCustomEventCalled.should.equal(true);
+        window.Leanplum.trackCustomName.should.equal('Purchase');
+
+        window.Leanplum.eventProperties[0].Sku.should.equal('12345');
+        window.Leanplum.eventProperties[0].Name.should.equal('iPhone 6');
+        window.Leanplum.eventProperties[0].Category.should.equal('Phones');
+        window.Leanplum.eventProperties[0].Brand.should.equal('iPhone');
+        window.Leanplum.eventProperties[0].Variant.should.equal('6');
+        window.Leanplum.eventProperties[0].Price.should.equal(400);
+        window.Leanplum.eventProperties[0].TotalAmount.should.equal(400);
+        window.Leanplum.eventProperties[0].CouponCode.should.equal('coupon-code');
+        window.Leanplum.eventProperties[0].Quantity.should.equal(1);
+
+        done();
+    });
+
+    it('it should set user identity if IdentityType = CustomerId', function(done) {
+        mParticle.forwarder.setUserIdentity('123abc', IdentityType.CustomerId);
+
+        window.Leanplum.userId.should.equal('123abc');
+
+        done();
+    });
+
+    it('it should set userId of type customerId if user identities passed to it include types customerId and email', function(done) {
+        window.Leanplum.userId.should.equal('customerId');
```

Comment:
Minor - test relies on previous test having executed. I think it's usually better if unit tests are independent of one another.